docs/cookbook/recipe_sortable_listing.rst

```diff
@@ -93,9 +89,10 @@ and use the default twig template provided in the ``pixSortableBehaviorBundle``
         ]);
 
 
-In order to add new routes for these actions we are also adding the following method::
+In order to add new routes for these actions we are also adding the following method:
+
+.. code-block:: php
```

Comment:
`::` is the same as `.. code-block:: php`

Reply:
Hmm, I think not really, implicitely yes, but per definition it is a code block, but the syntax highlighting _could_ be different.
We should be  explicit IMO

Is there sth available like php-cs-fixer for *rst files to provide some rules? 🤔

Reply:
and we should handle this like symfony docs

Reply:
https://github.com/symfony/symfony-docs/pull/9339

Reply:
Thank you for this info, didn't know that. 

But this is an interesting comment https://github.com/symfony/symfony-docs/pull/9339#discussion_r170520136

I am fine for any version, but we should not mix them

Reply:
Hmmm this is good to know. Can you please create a dev-kit PR? Because we cannot use the long syntax if we still have this sentence:

https://github.com/sonata-project/SonataAdminBundle/blob/5165c34555346849d60c120196b37c2b4d2dd50e/CONTRIBUTING.md#L117

Reply:
@javiereguilaz @weaverryan can you confirm my last comment?

If you have a look [here](https://sonata-project.org/bundles/admin/3-x/doc/reference/architecture.html#fields-definition), we have **no syntax-highlighting for PHP** and we are using this:
```rst
which stores instances of ``FieldDescriptionInterface``. Picking up on our previous
``PostAdmin`` class example::

    <?php
    // src/Admin/PostAdmin.php

    namespace App\Admin;

    use Sonata\AdminBundle\Admin\AbstractAdmin;
    use Sonata\AdminBundle\Datagrid\ListMapper;
    use Sonata\AdminBundle\Datagrid\DatagridMapper;
    use Sonata\AdminBundle\Form\FormMapper;
    use Sonata\AdminBundle\Show\ShowMapper;
    use Symfony\Bridge\Doctrine\Form\Type\EntityType;
    use Symfony\Component\Form\Extension\Core\Type\TextType;
    use App\Entity\User;

    class PostAdmin extends AbstractAdmin
    {
```

but [here ](https://sonata-project.org/bundles/admin/3-x/doc/reference/action_list.html#customizing-the-fields-displayed-on-the-list-page) we **have syntax-highlighting** and use:
 ```rst
Customizing the fields displayed on the list page
-------------------------------------------------

You can customize the columns displayed on the list through the ``configureListFields`` method.
Here is an example:

.. code-block:: php

    <?php
    // ...

    protected function configureListFields(ListMapper $listMapper)
    {
        $listMapper
```

And [here](https://sonata-project.org/bundles/admin/3-x/doc/getting_started/installation.html#enable-the-bundle) in the second PHP block we use:
```rst

.. note::
    If you are not using Symfony Flex, you should enable bundles in your
    ``AppKernel.php``.

.. code-block:: php

    // app/AppKernel.php

    // ...
    class AppKernel extends Kernel
    {
        public function registerBundles()
        {
            $bundles = [
                // ...

                // The admin requires some twig functions defined in the security
                // bundle, like is_granted. Register this bundle if it wasn't the case
                // already.
                new Symfony\Bundle\SecurityBundle\SecurityBundle(),

                // These are the other bundles the SonataAdminBundle relies on
                new Sonata\CoreBundle\SonataCoreBundle(),
                new Sonata\BlockBundle\SonataBlockBundle(),
                new Knp\Bundle\MenuBundle\KnpMenuBundle(),

                // And finally
                new Sonata\AdminBundle\SonataAdminBundle(),
            ];

            // ...
        }

        // ...
    }
```

So it points out, that if we want to have php-syntax-highlighting (at least for our theme on sonata-project.org)` we need to use `.. code-block:: php` directive **AND** `<?php`.

Reply:
I don’t think (not totally sure) that the theme affects parsing on this level - that’s more for the layout. I believe that, regardless of theme, the same Sphinx config would render the same markup for the code blocks, etc. But it *is* possible that you’re missing some CSS that Symfony.com has

Reply:
I think I found out why the syntax is highlighted on symfony.com and here only when using:
```
... coed-block:: php

    <?php
    echo "Hello World";
```

Have a look [here](http://mbless.de/blog/2015/03/02/php-syntax-highlighting-in-sphinx.html) and check the [symfony-docs conf.py](https://github.com/symfony/symfony-docs/blob/master/_build/conf.py#L103) afterwards.

This part is missing in our file:
```
# -- Settings for symfony doc extension ---------------------------------------------------

# enable highlighting for PHP code not between ``<?php ... ?>`` by default
lexers['markdown'] = TextLexer()
lexers['php'] = PhpLexer(startinline=True)
lexers['php-annotations'] = PhpLexer(startinline=True)
lexers['php-standalone'] = PhpLexer(startinline=True)
lexers['php-symfony'] = PhpLexer(startinline=True)
lexers['rst'] = RstLexer()
lexers['varnish3'] = CLexer()
lexers['varnish4'] = CLexer()
lexers['terminal'] = TerminalLexer()

config_block = {
    'apache': 'Apache',
    'markdown': 'Markdown',
    'nginx': 'Nginx',
    'rst': 'reStructuredText',
    'terminal': 'Terminal',
    'varnish3': 'Varnish 3',
    'varnish4': 'Varnish 4'
}

# use PHP as the primary domain
primary_domain = 'php'
```

I will try to get it work, and if, we should get rid of the php open tags.

Reply:
That last part seems indeed relevant: `primary_domain = 'php'`

Reply:
I thought this too, but here is the working change, the `primary_domain` is not important at all:
https://github.com/sonata-project/SonataAdminBundle/pull/5389